elastic/kibana
# Dashboard insights flyout with dashboard views

## Summary

close https://github.com/elastic/kibana/issues/183687

## Feature

- Implement dashboard view stats UI on top of usage counter that counts dashboard views for last 90 day and shows weekly histogram. 
  - (Even if there is not a lot of data, we still show it as a weekly histogram, so it can be pretty empty intially) 

![Screenshot 2024-08-15 at 13 00 11](https://github.com/user-attachments/assets/adeabf78-e3d3-4cfa-adc3-76a32ede595b)


## Implementation

### Server side
Dashboard plugin registers new routes to increase the view count and get stats. Routes are protected for users with dashboard access only. The implementation is located in `@kbn/content-management-content-insights-server` and internally uses usage counters. The retention is 90 days, so we can only show stats for last 90 days. 

 ### Client side

- Dashboard uses the client from `@kbn/content-management-content-insights-public` to increase the view count every time a user opens a dashboard. 
- TableListView opens the flyout from `@kbn/content-management-content-insights-public`to display the stats

## How to test

- For new views just open a dashboard and check that view stat is increased 
- For old views you can populate the usage counters with historic data. I used the following script: https://gist.github.com/Dosant/425042fcf75d5e40e5a46374f6234a54
